### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by hjanetzek**
Some issues found by valgrind when testing with raster sources

### CODE REVIEW

File: core/src/tangram.cpp [hunk 1]
```diff
@@ -109,7 +109,10 @@ Map::~Map() {
     // The unique_ptr to Impl will be automatically destroyed when Map is destroyed.
     impl->tileWorker.stop();
     impl->asyncWorker.reset();
-    impl->jobQueue.runJobs();
+
+    // Make sure other threads are stopped before calling stop()!
+    // All jobs will be executed immediately on add() afterwards.
+    impl->jobQueue.stop();
```
Comment (blair1618) [inline core/src/tangram.cpp]: @hjanetzek what's the bug that this fixes?

Comment (hjanetzek) [inline core/src/tangram.cpp]: Future bugs :) This should make things more robust though. E.g. in our current use Impl::renderState could be destructed before Impl::jobQueue without bugs.

The actual bug I was looking at was the static UrlWorkers still holding onto RasterTileTask-\>RasterSource-\>Textures where the texture disposer called into garbage.

Comment (blair1618) [inline core/src/tangram.cpp]: Yeah, the destruction process relies on the ordering of members right now which isn't ideal.

The case of holding onto tasks that invoke disposers does seem pretty risky, this change sounds like a good idea.

File: core/src/tile/tileWorker.cpp [hunk 2]
```diff
@@ -130,6 +130,8 @@ void TileWorker::stop() {
     for (auto& worker : m_workers) {
         worker->thread.join();
     }
+
+    m_queue.clear();
```
Comment (tallytalwar) [inline core/src/tile/tileWorker.cpp]: :+1:

**Pull request merged**